Change old moderation strikes to be displayed in a separate page

ClearlyClaire opened the pull request:
Fixes #17552

This changes the moderation strikes displayed on `/auth/edit` to be those from the past 3 months, and make all moderation strikes targeting the current user available in `/disputes`.

![image](https://user-images.githubusercontent.com/384364/154226916-bd60848b-c82b-4ebf-9d4d-52acbdd928ed.png)

ClearlyClaire committed:
Change old moderation strikes to be displayed in a separate page

Fixes #17552

This changes the moderation strikes displayed on `/auth/edit` to be those from
the past 3 months, and make all moderation strikes targeting the current user
available in `/disputes`.

Gargron commented on config/locales/en.yml:
```diff
@@ -1362,6 +1362,7 @@ en:
     profile: Profile
     relationships: Follows and followers
     statuses_cleanup: Automated post deletion
+    strikes: Moderation strikes
```
Would this be more understandable?

```suggestion
    strikes: Strikes against your account
```

nightpool replied:
maybe just "Moderation actions"? "Strikes" feels a little harsh.

Gargron replied:
They're called [strikes on YouTube](https://support.google.com/youtube/answer/2802032?hl=en) so people would be familiar with the concept

Gargron commented on app/views/disputes/strikes/index.html.haml:
```diff
@@ -0,0 +1,4 @@
+- content_for :page_title do
+  = t('settings.strikes')
+
```
```suggestion

%p= t('disputes.strikes.description_html', instance: Rails.configuration.x.local_domain)

```

"These are actions taken against your account and warnings that have been sent to you by the staff of %{instance}"

Gargron commented on config/navigation.rb:
```diff
@@ -20,9 +20,10 @@
+      s.item :password, safe_join([fa_icon('lock fw'), t('settings.account_settings')]), edit_user_registration_url, highlights_on: %r{/auth/edit|/settings/delete|/settings/migration|/settings/aliases|/settings/login_activities}
       s.item :two_factor_authentication, safe_join([fa_icon('mobile fw'), t('settings.two_factor_authentication')]), settings_two_factor_authentication_methods_url, highlights_on: %r{/settings/two_factor_authentication|/settings/otp_authentication|/settings/security_keys}
       s.item :authorized_apps, safe_join([fa_icon('list fw'), t('settings.authorized_apps')]), oauth_authorized_applications_url
+      s.item :strikes, safe_join([fa_icon('warning'), t('settings.strikes')]), disputes_strikes_url, if: -> { current_account.strikes.exists? }, highlights_on: %r{|^/disputes}
```
```suggestion
```

Instead of showing a whole menu item, I believe this page should be linked from under the "Account status" section on the Account settings page, similar to the login history.

Gargron commented on config/navigation.rb:
```diff
@@ -20,9 +20,10 @@
 
     n.item :security, safe_join([fa_icon('lock fw'), t('settings.account')]), edit_user_registration_url do |s|
-      s.item :password, safe_join([fa_icon('lock fw'), t('settings.account_settings')]), edit_user_registration_url, highlights_on: %r{/auth/edit|/settings/delete|/settings/migration|/settings/aliases|/settings/login_activities|^/disputes}
+      s.item :password, safe_join([fa_icon('lock fw'), t('settings.account_settings')]), edit_user_registration_url, highlights_on: %r{/auth/edit|/settings/delete|/settings/migration|/settings/aliases|/settings/login_activities}
```
```suggestion
      s.item :password, safe_join([fa_icon('lock fw'), t('settings.account_settings')]), edit_user_registration_url, highlights_on: %r{/auth/edit|/settings/delete|/settings/migration|/settings/aliases|/settings/login_activities|^/disputes}
```

Gargron reviewed (changes requested):
I agree with this change with these adjustments

Gargron commented:
Unfortunately I cannot submit suggestions in a review for lines and files that have not been touched in the PR.

ClearlyClaire committed:
Add short description of what the strikes page is for

ClearlyClaire committed:
Move link to list of strikes to “Account status” instead of navigation item

ClearlyClaire committed:
Normalize i18n file

ClearlyClaire committed:
Fix layout and styling of strikes link

ClearlyClaire committed:
Revert highlights_on regexp

ClearlyClaire committed:
Reintroduce account status summary

- this way, “Account status” is never empty
- account status is not necessarily bound to strikes, or recent strikes

Gargron reviewed (approved):
I'm not a huge fan of using the old text but it'll do
